backend/notes.go

```diff
@@ -0,0 +1,303 @@
+		// not accidentally miss any notes.
+
+		if err := account.Initialize(); err != nil {
+			return err
+		}
+		backend.log.WithField("code", account.Config().Config.Code).Info("Exporting name and notes of account")
+
+		accountName := account.Config().Config.Name
+		accountCode := account.Config().Config.Code
+
+		// We have unified accounts, but to be compatible with BIP-329, we export the same account
+		// name for each xpub in the unified account, so other wallets that import one of the
+		// sub-accounts will import the name.
+		//
+		// We don't export the name for ERC20 tokens, they are not "real" accounts (they are derived
+		// from the ETH account), and the user cannot modify the ERC20 token account name.
+		ethCoin, isETHCoin := account.Coin().(*eth.Coin)
+		isERC20 := isETHCoin && ethCoin.ERC20Token() != nil
+		if accountName != "" && !isERC20 {
+			for _, signingConfig := range account.Config().Config.SigningConfigurations {
+				entry := bip329Entry{
+					Type:  bip329TypeXpub,
+					Ref:   signingConfig.ExtendedPublicKey().String(),
+					Label: accountName,
+					// We don't technically need this for the import, as we assume no two accounts
+					// in the BitBoxApp contain the same xpub. Note that ERC20-token accounts have
+					// the same xpub as the parent ETH account, but they are auto-derived and do not
+					// exist in the persisted config where the account name is stored.
+					//
+					// We add it anyway just in case, as it can't hurt and is more explicit.
+					BitBoxApp: &bip329BitBoxApp{
+						CoinCode:    account.Config().Config.CoinCode,
+						AccountCode: account.Config().Config.Code,
+					},
+				}
+				if err := json.NewEncoder(writer).Encode(entry); err != nil {
+					return err
+				}
+			}
+		}
+
+		notesData := account.Notes().Data()
+		for txID, note := range notesData.TransactionNotes {
+			entry := bip329Entry{
+				Type:  bip329TypeTx,
+				Ref:   txID,
+				Label: util.TruncateString(note, notes.MaxNoteLen),
+				// We use this to aid the import back into the BitBoxApp, as the same txID can be in
+				// multiple accounts with different labels (e.g. and ERC20 token in the ERC20 token
+				// account as well as in the parent ETH account), and the origin label is not a good
+				// fit (see docstring of `bip329BitBoxApp`).
+				BitBoxApp: &bip329BitBoxApp{
+					CoinCode:    account.Config().Config.CoinCode,
+					AccountCode: accountCode,
+				},
+			}
+			if err := json.NewEncoder(writer).Encode(entry); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// ExportNotes exports the transaction and account labels of all accounts of all
+// connected/remembered keystores. Deactivated accounts are included in the export, except for
+// deactivated ERC-20 accounts. We export to a file using an extended version of BIP-329:
+// https://github.com/bitcoin/bips/blob/master/bip-0329.mediawiki
+func (backend *Backend) ExportNotes() error {
+	exportsDir, err := utilcfg.ExportsDir()
```

Comment:
Not really related to the PR, but I just realized that after playing a bit with my locale settings I don't have a 'Download' folder anymore. I have the Italian equivalent 'Scaricati' instead and this causes the export to fail. Should we change to general export behavior so that the folder will be created if unavailable?

Reply:
Hmm yeah that would make sense, in a separate PR. Is there maybe an ENV var that points to the right folder?